lib/game.rb

```diff
@@ -1,151 +1,157 @@
-#game class file
-require './lib/cell'
-require './lib/board'
-require './lib/ship'
+class Game
+  attr_reader :cruiser, :submarine, :computer, :player_cells_fired_upon
+  attr_accessor :player_board
 
- class Game
-
-  attr_reader :player, :cruiser, :submarine
-  attr_accessor :computer_board
-
-  def initialize(player)
-    @player = player
-    @computer_board = Board.new
-    @player_board = Board.new
+  def initialize
+    @player_board ||= Board.new
     @cruiser = Ship.new("Cruiser", 3)
     @submarine = Ship.new("Submarine", 2)
-
+    @computer = Computer.new
+    @player_cells_fired_upon = []
   end
 
-  def user_places_ships(ship, coordinates)
-    loop do
-      p "which ship would you like to place? => "
-        ship = gets.chomp
-      p "enter coordinates => "
-        coordinates = gets.chomp
-      if @player_board.valid_placement?(ship, coordinates)
-          @player_board.place(ship, coordinates)
-          break
-      else
-        p "Those are invalid cooridnates. Please try again!"
-      end
+  def main_menu
+    puts "Welcome to BATTLESHIP \n" +
+    "Enter p to play. Enter q to quit. \n"
+    reply = gets.chomp
+    if reply == "p"
+      setup
+    elsif
+      reply == "q"
+      puts "Goodbye"
+    else
+      main_menu
     end
   end
 
-  def random_coordinates_cruiser
-    random_number = rand(1..10)
-
-    if random_number.even?
-      coordinates_array = []
-      x_axis = ("A".."D").to_a
-      x_axis = x_axis[rand(0..3)]
-      coord = x_axis
-
-      y_axis = ("1".."4").to_a
-      y_axis.each do |y|
-          coordinates_array << coord + y
-        end
+  def setup
+    computer.place_cruiser
+    computer.place_submarine
+    puts "I have laid out my ships on the grip. \n"+
+    "You now need to lay out your two ships. \n"+
+    "The Cruiser is three units long and the Submarine is two units long. \n"
+    @player_board.render
+    "Enter the squares for the Cruiser (3 spaces): "
+    user_input = gets.chomp
+    coordinates = user_input.upcase.gsub(/[^[:word:]\s]/,'').scan(/../)
```

Comment:
found this on the interwebs and pieced it together so if someone inputs "a1a2a3" it will make it into an array and split the coordinates...but need to fix it so it works with commas and spaces (which it did at one point and then I got lost in the weeds!)